app/controllers/houseConsignment/index/items/document/AddAnotherDocumentController.scala

```diff
@@ -67,7 +67,10 @@ class AddAnotherDocumentController @Inject() (
                   .onPageLoad(arrivalId, mode, houseConsignmentIndex, itemsIndex, viewModel.nextIndex)
               )
             case false =>
-              Redirect(controllers.routes.HouseConsignmentController.onPageLoad(arrivalId, houseConsignmentIndex))
+              Redirect(
+                controllers.houseConsignment.index.items.routes.AddAdditionalReferenceYesNoController
+                  .onPageLoad(arrivalId, houseConsignmentIndex, itemsIndex, mode)
+              )
```

Comment:
Hmm...I don't like the thought of being on the cross-check page, clicking the add/remove link for item 1 documents, clicking No then having to go through the rest of the item journey

Reply:
We were in a similar predicament with seals and goods references. I hope we won't have to do something similar but we might (i.e. track two modes). Then we would be able to distinguish between these 3 scenarios:

1. adding an item, adding/removing a document
2. changing an item, adding/removing a document
3. changing an item, changing a document

Reply:
This would work, but it would be an enormous change. Every page with an item index would need 8 routes. Like we have for seals and goods references
e.g.https://github.com/hmrc/manage-transit-movements-unloading-frontend/blob/main/conf/app.transportEquipment.index.seals.routes